src/cmake/axom-config.cmake.in

```diff
@@ -0,0 +1,131 @@
+#
+# SPDX-License-Identifier: (BSD-3-Clause)
+#------------------------------------------------------------------------------
+
+cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+
+@PACKAGE_INIT@
+
+if(NOT AXOM_FOUND)
+
+  #----------------------------------------------------------------------------
+  # Set version and paths
+  #----------------------------------------------------------------------------
+  
+  set(AXOM_VERSION       "@AXOM_VERSION_FULL@")
+  set(AXOM_VERSION_MAJOR "@AXOM_VERSION_MAJOR@")
+  set(AXOM_VERSION_MINOR "@AXOM_VERSION_MINOR@")
+  set(AXOM_VERSION_PATCH "@AXOM_VERSION_PATCH@")
+  
+  set(AXOM_INSTALL_PREFIX "@AXOM_INSTALL_PREFIX@")
+  set(AXOM_INCLUDE_DIRS "${AXOM_INSTALL_PREFIX}/include")
+
+  #----------------------------------------------------------------------------
+  # Set user configuration options and features
+  #----------------------------------------------------------------------------
+
+  # Language features
+  set(AXOM_ENABLE_FORTRAN     "@ENABLE_FORTRAN@")
+  set(AXOM_USE_CUDA           "@AXOM_USE_CUDA@")
+  set(AXOM_USE_MPI            "@AXOM_USE_MPI@")
+  set(AXOM_USE_OPENMP         "@AXOM_USE_OPENMP@")
+
+  # Axom components
+  set(AXOM_ENABLE_LUMBERJACK  "@AXOM_ENABLE_LUMBERJACK@")
+  set(AXOM_ENABLE_MINT        "@AXOM_ENABLE_MINT@")
+  set(AXOM_ENABLE_PRIMAL      "@AXOM_ENABLE_PRIMAL@")
+  set(AXOM_ENABLE_QUEST       "@AXOM_ENABLE_QUEST@")
+  set(AXOM_ENABLE_SIDRE       "@AXOM_ENABLE_SIDRE@")
+  set(AXOM_ENABLE_SLAM        "@AXOM_ENABLE_SLAM@")
+  set(AXOM_ENABLE_SLIC        "@AXOM_ENABLE_SLIC@")
+  set(AXOM_ENABLE_SPIN        "@AXOM_ENABLE_SPIN@")
+
+  # Axom built-in TPLs
+  set(AXOM_USE_CLI11          "@AXOM_USE_CLI11@")
+  set(AXOM_USE_FMT            "@AXOM_USE_FMT@")
+  set(AXOM_USE_SPARSEHASH     "@AXOM_USE_SPARSEHASH@")
+
+  # Axom external TPLs
+  set(AXOM_USE_CONDUIT        "@AXOM_USE_CONDUIT@")
+  set(AXOM_USE_HDF5           "@AXOM_USE_HDF5@")
+  set(AXOM_USE_MFEM           "@AXOM_USE_MFEM@")
+  set(AXOM_USE_RAJA           "@AXOM_USE_RAJA@")
+  set(AXOM_USE_SCR            "@AXOM_USE_SCR@")
+  set(AXOM_USE_UMPIRE         "@AXOM_USE_UMPIRE@")
+
+  #----------------------------------------------------------------------------
+  # Bring in required  dependencies for this axom configuration
+  #----------------------------------------------------------------------------
+  include(CMakeFindDependencyMacro)
+
+  # conduit
+  if(AXOM_USE_CONDUIT)
+    set(AXOM_CONDUIT_DIR  "@CONDUIT_DIR@")
+    if(NOT CONDUIT_DIR) 
+      set(CONDUIT_DIR ${AXOM_CONDUIT_DIR}) 
+    endif()
+    find_dependency(Conduit REQUIRED NO_DEFAULT_PATH PATHS "${CONDUIT_DIR}/lib/cmake")
+  endif()
+
+  # hdf5
+  if(AXOM_USE_HDF5)
+    set(AXOM_HDF5_DIR     "@HDF5_DIR@")
+    # Note: Targets not currently imported
+  endif()
+
+  # mfem
+  if(AXOM_USE_MFEM)
+    set(AXOM_MFEM_DIR     "@MFEM_DIR@")
+    # Note: Targets not currently imported
+  endif()
+
+  # raja
+  if(AXOM_USE_RAJA)
```

Comment:
SInce we are doing more or less the same thing for each TPL, we may want to consider having a macro, `axom_import_dependencies(...)`. Perhaps, it could be also made general enough to go in BLT.

Reply:
Thanks @gzagaris.

At this point, I am not sure there is enough custom code to warrant a separate macro. These three happen to look similar, but other TPLs (e.g. mfem or hdf5, if we want/need to export it) would likely look different.

Reply:
That sounds fair.